graph/task.go

```diff
@@ -155,18 +144,6 @@ func (t *Task) initialize() error {
 		t.TotalTimeout = t.StepTimeout
 	}
 
-	if t.StepTimeout < minStepTimeoutInSeconds {
```

Comment:
curious why we are allowing anything to set a minTimeout of a value less than 10 seconds.

Reply:
step/task timeout isn't provided in most task-based yamls out of the box so we need this default initialization logic. What's the intent behind deleting this block?

Reply:
same, not sure the intent of removing the min/max stuff. But the rest LGTM.

Reply:
I think we still want these min/max bounds. Rest LGTM.

Reply:
We have several cases which have different timeout settings. The services also has timeout control. It's tricky to sync all of them. It becomes hard to reason which timeout is hit when uses report "context deadline exceeded" error.
My suggestion is 
1) Keep acb logic simple, less default 
2) Give users full control, don't override it unless it truly impacts the functionality and it's hard for users to figure out. For example, if users set negative timeout value, we should override it or report error. If users want to timeout 5 seconds on a particular step, I don't see why we need to override it.